SecurityPkg: Remove PcdFirmwareDebuggerInitialized

VivianNK opened the pull request:
## Description

Removed PCD in favor of using DeviceStateLib for
storing firmware debugger enablement status.

This PR is waiting on the removal of the Pcd's use here https://github.com/microsoft/mu_tiano_plus/pull/342
For details on how to complete these options and their meaning refer to [CONTRIBUTING.md](https://github.com/microsoft/mu/blob/HEAD/CONTRIBUTING.md).

- [x] Impacts functionality?
- [ ] Impacts security?
- [x] Breaking change?
- [ ] Includes tests?
- [ ] Includes documentation?

## How This Was Tested

Local CI

## Integration Instructions

Remove any usage of PcdFirmwareDebuggerInitialized.

os-d commented on SecurityPkg/Tcg/Tcg2Dxe/Tcg2Dxe.inf:
```diff
@@ -97,15 +97,14 @@
 [Protocols]
   gEfiTcg2ProtocolGuid                               ## PRODUCES
-  gMuTcg2ProtocolExGuid                              ## PRODUCES # MU_CHANGE - Add a new protocol to support Log-only events.
+  gMuTcg2ProtocolExGuid                              ## PRODUCES # MU_CHANGE - Add a new protocol to support Log-only events.
```
What changed here?

os-d reviewed (approved):
I wouldn't mark this as non-functional, I do think this has a functional change, that the debug string will not be measured if you aren't using device state.

makubacki commented:
@VivianNK, you're going to merge this after support is added to use DeviceStateLib, right?

VivianNK commented:
> @VivianNK, you're going to merge this after support is added to use DeviceStateLib, right?

Yes @makubacki, it's waiting on this PR: https://github.com/microsoft/mu_tiano_plus/pull/342 . Added to the PR description.

VivianNK commented on SecurityPkg/SecurityPkg.dec:
```diff
@@ -446,13 +446,6 @@
   # @Prompt Enable hardware method of asserting physical presence.
   gEfiSecurityPkgTokenSpaceGuid.PcdPhysicalPresenceHwEnable|TRUE|BOOLEAN|0x00010005
 
-[PcdsFixedAtBuild, PcdsPatchableInModule, PcdsDynamic, PcdsDynamicEx]
```
TODO: comment out with MU_CHANGE tags

VivianNK committed:
SecurityPkg: Remove PcdFirmwareDebuggerInitialized

Removed PCD in favor of using DeviceStateLib for
storing firmware debugger enablement status.

Signed-off-by: Vivian Nowka-Keane <vnowkakeane@microsoft.com>
